core/src/main/scala/org/apache/spark/util/DependencyUtils.scala

```diff
@@ -15,22 +15,158 @@
+      // in a excluded list.
+      val exclusionList = groupedParams.get("exclude").map { params =>
+        params.map(_._2).flatMap { excludeString =>
+          val excludes = excludeString.split(",")
+          if (excludes.map(_.split(":")).exists(token =>
+            token.length != 2 || StringUtils.isBlank(token(0)) || StringUtils.isBlank(token(1)))) {
+            throw new URISyntaxException(uri.toString, "Invalid exclude string: " +
+              "expected 'org:module,org:module,..', found " + excludeString)
+          }
+          excludes
+        }.mkString(",")
+      }.getOrElse("")
+
+      val invalidParams = groupedParams
+        .filter(entry => !Seq("transitive", "exclude").contains(entry._1))
+        .keys.toArray.sorted
+      if (invalidParams.nonEmpty) {
+        logWarning(
+          s"Invalid parameters `${invalidParams.mkString(",")}` found in URI query `$uriQuery`.")
+      }
+
+      groupedParams.foreach { case (key: String, values: Array[(String, String)]) =>
+        if (key != "transitive" || key != "exclude") {
+          logWarning("Invalid parameter")
+        }
+      }
+
+      (transitive, exclusionList)
+    }
+  }
+
+  /**
+   * Download Ivy URIs dependency jars.
+   *
+   * @param uri Ivy uri need to be downloaded. The URI format should be:
+   *              `ivy://group:module:version[?query]`
+   *            Ivy URI query part format should be:
+   *              `parameter=value&parameter=value...`
+   *            Note that currently ivy URI query part support two parameters:
+   *             1. transitive: whether to download dependent jars related to your ivy URL.
+   *                transitive=false or `transitive=true`, if not set, the default value is false.
+   *             2. exclude: exclusion list when download ivy URL jar and dependency jars.
+   *                The `exclude` parameter content is a ',' separated `group:module` pair string :
+   *                `exclude=group:module,group:module...`
+   * @return Comma separated string list of jars downloaded.
+   */
+  def resolveMavenDependencies(uri: URI): Seq[String] = {
+    try {
+      val ivyProperties = DependencyUtils.getIvyProperties()
+      val authority = uri.getAuthority
+      if (authority == null) {
+        throw new URISyntaxException(
+          uri.toString, "Invalid url: Expected 'org:module:version', found null")
+      }
+      if (authority.split(":").length != 3) {
+        throw new URISyntaxException(
+          uri.toString, "Invalid url: Expected 'org:module:version', found " + authority)
+      }
+
+      val (transitive, exclusionList) = parseQueryParams(uri)
+
+      resolveMavenDependencies(
+        transitive,
+        exclusionList,
+        authority,
+        ivyProperties.repositories,
+        ivyProperties.ivyRepoPath,
+        Option(ivyProperties.ivySettingsPath)
+      ).split(",")
+    } catch {
+      case e: URISyntaxException =>
```

Comment:
Which library API throws this exception? Could you minimize the try-catch block?

Reply:
> Which library API throws this exception? Could you minimize the try-catch block?

Use `IllegalArgumentException ` directly  , remove try catch